doc/development_process/api_lifecycle.rst

```diff
@@ -77,16 +77,70 @@ An API can be declared ``stable`` after fulfilling the following requirements:
 - Stable APIs can get backward compatible updates, bug fixes and security fixes
   at any time.
 
+Introducing incompatible changes
+================================
+
+A stable API, as described above strives to remain backwards-compatible through
+its life-cycle. There are however cases where fulfilling this objective prevents
+technical progress or is simply unfeasible without unreasonable burden on the
+maintenance of the API and its implementation(s).
+
+In order to restrict and control the introduction of a change that breaks the
+promise of backwards compatibility the following steps must be followed whenever
+such a change is considered necessary in order to accept it in the project:
+
+#. An RFC issue must be opened on GitHub with the following content:
```

Comment:
Don't we have an RFC template that gets automatically populated when you open an issue of that type?

https://github.com/zephyrproject-rtos/zephyr/issues/new?assignees=&labels=RFC&template=rfc---proposal.md&title=

Reply:
Yes, I see that template too, and it is asking for different content than noted below.  I suspect, if agreed to, the template should be updated to reflect this documentation, and the documentation should be updated to say to click on the RFC/Proposal template when submitting the issue.

Reply:
The RFC template is for general RFCs, we do not want to make it API specific.

Reply:
I think the idea was to use this template but with extra contents. We can extend existing template or create new - API specific. I would opt for extensions to existing one.

Reply:
The RFC template ought to be able to cover any RFC

Reply:
Fair enough, so what I will do is to adapt the list below to match into the current RFC template we have.